packages/salesforcedx-vscode-core/src/orgBrowser/metadataOutlineProvider.ts

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (c) 2019, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import { isNullOrUndefined } from 'util';
+import * as vscode from 'vscode';
+import { hasRootWorkspace, OrgAuthInfo } from '../util';
+import { BrowserNode, NodeType } from './index';
+
+export class MetadataOutlineProvider
+  implements vscode.TreeDataProvider<BrowserNode> {
+  private defaultOrg: string | undefined;
+
+  private _onDidChangeTreeData: vscode.EventEmitter<
+    BrowserNode | undefined
+  > = new vscode.EventEmitter<BrowserNode | undefined>();
+  public readonly onDidChangeTreeData: vscode.Event<
+    BrowserNode | undefined
+  > = this._onDidChangeTreeData.event;
+
+  constructor() {}
+
+  public async refresh(): Promise<void> {
+    await this.getDefaultUsernameOrAlias();
+    this._onDidChangeTreeData.fire();
+  }
+
+  public getTreeItem(element: BrowserNode): vscode.TreeItem {
+    return element;
+  }
+
+  public getChildren(element?: BrowserNode): Promise<BrowserNode[]> {
+    if (isNullOrUndefined(element)) {
+      if (!isNullOrUndefined(this.defaultOrg)) {
+        const org = new BrowserNode(this.defaultOrg, NodeType.Org);
+        return Promise.resolve([org]);
+      } else {
+        return Promise.resolve([]);
```

Comment:
Shouldn't this be a reject ? Are we going to show a message to the user that no org/defaultusername is available for use ?

Reply:
I was playing around with actually displaying a message when there is no org set, in which case we would actually be returning a node with the label set to a message and the collapsible state set to none. In that case, I think it would remain Promise.resolve

Reply:
Cool, thanks for clarifying